docs/DeveloperGuide.md

```diff
@@ -1,34 +1,600 @@
+
+4. `PatientParser#parseAddCommand()` will first validate the attributes and then create an object `AddPatientCommand` 
+if the attributes are valid. 
+
+5. `HappyPills` then calls `AddPatientCommand.execute()` to execute the command. 
+
+6. In `AddPatientCommand.execute()`, a patient is added and the display message is returned.
+
+#### 4.1.2 Edit Patient Details
+
+The user is able to edit the details of a patient in the list of patients currently in the program. The command: 
+
+    edit patient S7777777Z /a School
+    
+will edit the patient's allergies to `School`. 
+
+**Implementation** 
+
+![Edit Patient Sequence Diagram](images/EditPatientSequenceDiagram.jpg)
+
+The following steps explains the sequence of events: 
+
+1. The user enters `edit patient S7777777Z /a School` 
+
+2. `HappyPills` calls `Parser#parse()` which then calls `PatientParser#parse()`.
+
+3. `PatientParser#parse()` will then create an object `EditPatientCommand`. 
+
+4. `HappyPills` then calls `EditPatientCommand.execute()` method to execute the command. 
+
+5. In `EditPatientCommand.execute()`, the given attribute of the patient is edited and the display message is returned. 
+
+#### 4.1.3. Delete Patient Details 
+
+The user can delete a patient from the list of patients currently in the program. The command: 
+
+    delete patient S7777777Z 
+    
+will delete the patient with NRIC `S7777777Z`, if found. 
+
+**Implementation** 
+
+![Delete Patient Sequence Diagram](images/DeletePatientSequenceDiagram.jpg)
+
+The following steps explains the sequence of events: 
+
+1. The user enters `delete patient S7777777Z`. 
+
+2. `HappyPills` calls `Parser#parse()` which then calls `PatientParser#parse()`.
+
+3. `PatientParser#parse()` will then create an object `DeletePatientCommand`. 
+
+4. `HappyPills` then calls `DeletePatientCommand.execute()` to execute the command. 
+
+5. In `DeletePatientCommand.execute()`, the respective patient is deleted and the display message is returned. 
+
+#### 4.1.4. List Patients
+
+The user is able to get a list of all the patients currently in the program.
+The command: 
+    
+    list patient
+    
+will list all the appointments in the `PatientMap`. 
+
+**Implementation** 
+
+The following steps explains the sequence of events: 
+
+1. The user enters `list patient`.
+
+2. `HappyPills` calls `Parser#parse()` which then calls `PatientParser#parse()`.
+
+3. `APatientParser#parse()` will create an object `ListPatientCommand`.
+
+4. `HappyPills` then calls `ListPatientCommand.execute()` to execute the method. 
+
+5. In `ListPatientCommand.execute()`, it calls `TextUi.getPatientList()` which is then returned as the display message. 
+
+#### 4.1.5. Retrieve Patient Details 
+
+The user can retrieve a patient's details. The command: 
+
+    get patient S7777777Z
+    
+will retrieve the details of the patient with NRIC `S7777777Z`. 
+
+**Implementation** 
+
+![Get Patient Sequence Diagram](images/GetPatientSequenceDiagram.jpg)
```

Comment:
Like in `images/AddPatientSequenceDiagram.jpg`, I suggest shortening the activation bars of the methods, as a method cannot remain active after it has returned.

From the CS2113/T website:
![bug1](https://user-images.githubusercontent.com/35903974/77999573-77fa3980-7365-11ea-8274-f6f23cffa7b2.png)